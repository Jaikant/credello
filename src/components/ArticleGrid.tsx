import * as React from 'react';
import styled from 'styled-components';
import { device } from 'libs/device';
import { Text } from 'components/Typography';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import { FullWidthWrapper } from 'components/LayoutComponents';

const ComponentYPadding = styled(FullWidthWrapper)`
  padding-left: 10px;
  padding-top: ${({ theme }) => theme.spacing.padding13};
  padding-bottom: ${({ theme }) => theme.spacing.padding22};
`;

const WrapperArticleGrid = styled(ComponentYPadding)`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(4, calc(50% - 30px));
  grid-template-rows: minmax(120px, 1fr);
  overflow: scroll;
  background: rgb(243, 244, 253);
  scroll-snap-type: x proximity;
  &::-webkit-scrollbar {
    display: none;
  }
  scrollbar-width: none;
  -ms-overflow-style: none;
  @media ${device.tablet} {
    grid-gap: 32px;
    grid-template-columns: repeat(4, calc(20%));
    grid-template-rows: minmax(120px, 1fr);
    padding: 24px 24px;
    margin: 0px -24px;
  }
  @media ${device.laptop} {
    grid-gap: 64px;
    grid-template-columns: repeat(4, calc(20% - 40px));
    grid-template-rows: minmax(120px, 1fr);
    padding: 32px 128px;
    margin: 0px -128px;
  }
`;

const StyledCard = styled(Card)`
  &.MuiPaper-root {
    background: inherit;
    box-shadow: none;
    border-radius: 0px;
  }
`;

const StyledCardMedia = styled(CardMedia)`
  height: 71px;
  border-radius: 4px;
`;

const StyledCardContent = styled(CardContent)`
  &.MuiCardContent-root {
    overflow: visible;
    padding: 0;
    padding-top: 14px;
  }
`;

function MediaCard(props) {
  const { image, title } = props;
  return (
    <StyledCard>
      <CardActionArea>
        <StyledCardMedia
          image={require(`../assets/images/${image}`)}
          title={title}
        />
        <StyledCardContent>
          <Text mb="8px" lh="20px" color="#6c7f87">
            {title}
          </Text>
        </StyledCardContent>
      </CardActionArea>
    </StyledCard>
  );
}

MediaCard.defaultProps = {
  image: 'featured.png',
  category: 'DEBT 101',
  updated: ' | Mar 19, 2020',
  title: `Whats a Normal Amount of Debt... and What's Too Much?`,
  content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
  feel anxious. Read on to see what debt looks like for an...`,
};

const ArticleGrid = ({ cards }) => {
  return (
    <WrapperArticleGrid>
      {cards.map((c, index) => {
        return <MediaCard key={index} {...c} />;
      })}
    </WrapperArticleGrid>
  );
};

ArticleGrid.defaultProps = {
  cards: [
    {
      image: 'loanCredit.png',
      category: 'DEBT 101',
      updated: ' | Mar 19, 2020',
      title: `How to get a personal loan with a credit card?`,
      content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
      feel anxious. Read on to see what debt looks like for an...`,
    },
    {
      image: 'refinance.png',
      category: 'DEBT 101',
      updated: ' | Mar 19, 2020',
      title: `Should I refinance during the Covid-19 outbreak?`,
      content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
      feel anxious. Read on to see what debt looks like for an...`,
    },
    {
      image: 'creditDebt.png',
      category: 'DEBT 101',
      updated: ' | Mar 19, 2020',
      title: `How to get out of credit card debt fast?`,
      content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
      feel anxious. Read on to see what debt looks like for an...`,
    },
    {
      image: 'featured.png',
      category: 'DEBT 101',
      updated: ' | Mar 19, 2020',
      title: `Whats a Normal Amount of Debt... and What's Too Much?`,
      content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
      feel anxious. Read on to see what debt looks like for an...`,
    },
  ],
};

export default ArticleGrid;
