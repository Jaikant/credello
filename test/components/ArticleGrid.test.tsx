import React from 'react';
import { render } from '../testUtils';
import ArticleGrid from '../../src/components/ArticleGrid';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<ArticleGrid />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
