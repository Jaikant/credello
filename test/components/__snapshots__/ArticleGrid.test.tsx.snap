// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c4 {
  font-family: 'Roboto';
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
  webkit-font-smoothing: antialiased;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  margin-left: 0;
  color: #6c7f87;
  text-align: initial;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  margin-right: -22px;
  margin-left: -22px;
  padding-right: 22px;
  padding-left: 22px;
  padding-left: 10px;
  padding-top: 13px;
  padding-bottom: 22px;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(4,calc(50% - 30px));
  grid-template-rows: minmax(120px,1fr);
  overflow: scroll;
  background: rgb(243,244,253);
  -webkit-scroll-snap-type: x proximity;
  -moz-scroll-snap-type: x proximity;
  -ms-scroll-snap-type: x proximity;
  scroll-snap-type: x proximity;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.c0::-webkit-scrollbar {
  display: none;
}

.c1.MuiPaper-root {
  background: inherit;
  box-shadow: none;
  border-radius: 0px;
}

.c2 {
  height: 71px;
  border-radius: 4px;
}

.c3.MuiCardContent-root {
  overflow: visible;
  padding: 0;
  padding-top: 14px;
}

@media (min-width:768px) {

}

@media (min-width:1224px) {

}

@media (min-width:768px) {
  .c0 {
    margin-right: -24px;
    margin-left: -24px;
    padding-right: 24px;
    padding-left: 24px;
  }
}

@media (min-width:1224px) {
  .c0 {
    margin-right: -128px;
    margin-left: -128px;
    padding-right: 128px;
    padding-left: 128px;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-gap: 32px;
    grid-template-columns: repeat(4,calc(20%));
    grid-template-rows: minmax(120px,1fr);
    padding: 24px 24px;
    margin: 0px -24px;
  }
}

@media (min-width:1224px) {
  .c0 {
    grid-gap: 64px;
    grid-template-columns: repeat(4,calc(20% - 40px));
    grid-template-rows: minmax(120px,1fr);
    padding: 32px 128px;
    margin: 0px -128px;
  }
}

<div
    class="LayoutComponents__FullWidthWrapper-sc-3dcc8x-1 ArticleGrid__ComponentYPadding-v8krbb-0 c0"
  >
    <div
      class="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c2"
          style="background-image: url(test-file-stub);"
          title="How to get a personal loan with a credit card?"
        />
        <div
          class="MuiCardContent-root c3"
        >
          <div
            class="c4"
            color="#6c7f87"
          >
            How to get a personal loan with a credit card?
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c2"
          style="background-image: url(test-file-stub);"
          title="Should I refinance during the Covid-19 outbreak?"
        />
        <div
          class="MuiCardContent-root c3"
        >
          <div
            class="c4"
            color="#6c7f87"
          >
            Should I refinance during the Covid-19 outbreak?
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c2"
          style="background-image: url(test-file-stub);"
          title="How to get out of credit card debt fast?"
        />
        <div
          class="MuiCardContent-root c3"
        >
          <div
            class="c4"
            color="#6c7f87"
          >
            How to get out of credit card debt fast?
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root c1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c2"
          style="background-image: url(test-file-stub);"
          title="Whats a Normal Amount of Debt... and What's Too Much?"
        />
        <div
          class="MuiCardContent-root c3"
        >
          <div
            class="c4"
            color="#6c7f87"
          >
            Whats a Normal Amount of Debt... and What's Too Much?
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
